### PR TITLE
Update to clojure 1.9

### PR DESCRIPTION
**Pull request by gigasquid**
Updating to clojure 1.9

Requires a small update in think.image https://github.com/thinktopic/think.image/pull/10 - once the pull request is merged, the version numbers in this pull request can be updated.

Tested all the examples and they seem to work fine.

The only area that I didn't test, because I don't have all the local deps, are the importers. They would still need to be checked out

### CODE REVIEW

File: experiment/project.clj [hunk 1]
```diff
@@ -10,7 +10,7 @@
                  [thinktopic/think.gate "0.1.3"]
                  ;;This had better precisely match the version of figwheel that think.gate uses
                  ;;Tried with 1.9.XXX and had odd unexplainable failures.
```
Comment (gigasquid) [inline experiment/project.clj]: I didn't see any instability during my testing - but would benefit from another pair of eyes

File: examples/catsdogs-classification/src/catsdogs/training.clj [hunk 2]
```diff
@@ -85,11 +85,12 @@
...
-  (classification/perform-experiment
-    (initial-description image-size image-size num-classes)
-                                   train-ds
-                                   test-ds
-                                   observation->image
-                                   class-mapping
-                                   {})
-)
\ No newline at end of file
+  (let [listener (classification/create-listener observation->image
```
Comment (gigasquid) [inline examples/catsdogs-classification/src/catsdogs/training.clj]: moved to new `perfom-expermint` syntax with listeners

File: examples/catsdogs-classification/project.clj [hunk 3]
```diff
@@ -1,20 +1,18 @@
...
+(defproject catsdogs-classification "0.9.12-SNAPSHOT"
   :description "An example of using experiment/classification on mnist."
-  :dependencies [[org.clojure/clojure "1.8.0"]
+  :dependencies [[org.clojure/clojure "1.9.0-alpha17"]
                  [thinktopic/experiment "0.9.12-SNAPSHOT"]
                  [org.clojure/tools.cli "0.3.5"]
 
                  ; to manipulate images
-                 [thinktopic/think.image "0.4.8"]
+                 [thinktopic/think.image "0.4.11-SNAPSHOT"]
```
Comment (gigasquid) [inline examples/catsdogs-classification/project.clj]: Will need to be updated with newer version of think.image without spec error

File: examples/mnist-classification/project.clj [hunk 4]
```diff
@@ -1,9 +1,9 @@
 (defproject mnist-classification "0.9.12-SNAPSHOT"
   :description "An example of using experiment/classification on mnist."
-  :dependencies [[org.clojure/clojure "1.8.0"]
+  :dependencies [[org.clojure/clojure "1.9.0-alpha17"]
                  [thinktopic/experiment "0.9.12-SNAPSHOT"]
                  [org.clojure/tools.cli "0.3.5"]
-                 [thinktopic/think.image "0.4.8"]
+                 [thinktopic/think.image "0.4.11-SNAPSHOT"]
```
Comment (gigasquid) [inline examples/mnist-classification/project.clj]: Will need to be updated with newer version of think.image without spec error

File: examples/catsdogs-classification/src/catsdogs/training.clj [hunk 5]
```diff
@@ -1,5 +1,5 @@
 (ns catsdogs.training
-  (require [clojure.java.io :as io]))
+  (:require [clojure.java.io :as io]))
```
Comment (harold) [inline examples/catsdogs-classification/src/catsdogs/training.clj]: Ha! Nice catch here.

**Comment by harold** [comment]
Hi @gigasquid, thanks for this. @cnuernber is out this week, so don't panic if there isn't movement on this right away. We have internal projects that are moving to 1.9 so this is a priority for us as well. We appreciate your interest and effort as always.

File: examples/catsdogs-classification/src/catsdogs/training.clj (hunk 5 above)
Comment (gigasquid) [inline examples/catsdogs-classification/src/catsdogs/training.clj]: it was spec that caught it not me 😄

**Pull request merged**